File: day5.py
#!/usr/bin/env python3

from hashlib import md5

password = [',', ',', ',', ',', ',', ',', ',', ',']
contents = 'wtnhxymk'
total = 0

# setup to work for part 2
while ',' in password:
    temp = contents + str(total)
    if md5(temp.encode('utf-8')).hexdigest()[:5] == '00000':
        hashed = md5(temp.encode('utf-8')).hexdigest()
        if hashed[5] in '01234567':
            spot = int(hashed[5])
            if password[spot] == ',':
                password[spot] = hashed[6]
    total += 1



print(''.join(password))

File: day5.pl
#!/usr/bin/env perl

# perl solution for part 1 of day 5 made after overwriting part 1 in python
use strict;
use warnings;

use Digest::MD5 qw/md5_hex/;

my $input = 'wtnhxymk';
my $count = 0;

my $ans = '';

my $hash = md5_hex($input, $count);

while (length ($ans) < 8) {
  if (substr($hash, 0, 5) eq '00000') {
    $ans .= substr($hash, 5, 1)
  }

  $count++;
  $hash = md5_hex($input, $count);
}

print $ans;
